### PR TITLE
Shared UI for wordads

### PR DESCRIPTION
**Pull request by gravityrail**
This patch creates an independent, shared UI for WordAds, inspired by Jetpack Search. 

wordads-basic-ui

#### Changes proposed in this Pull Request:

* New shared UI for WordAds, disabled by default in the production build of Jetpack (for now)
* Update Search UI to use new shared `Gridicon` component

#### Jetpack product discussion

The goals of centralising and rewriting the UI are as follows:
- Single UI codebase for Jetpack and Calypso (current Calypso implementation is very out-of-sync with Jetpack implementation)
- Foundation for standalone WordAds plugin, if we go that route
- Enables users to view all their WordAds information (placements, payments, etc) in one place, rather than having to go to Calypso to do some of those things
- Much more modern experience using up-to-date styleguides, fonts, etc.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Edit your wp-config.php and add the line `define( 'ENABLE_WORDADS_SHARED_UI', true );`. By default this will be in `tools/docker/wordpress/wp-config.php`
* Ensure Jetpack is connected (the UI doesn't currently detect connection status or prompt to connect Jetpack)
* Go to https://goldsounds.jurassic.tube/wp-admin/admin.php?page=jetpack-wordads
* Check it looks like the screenshot at the top of this description.
* Don't worry if the toggle doesn't actually work.

### CODE REVIEW

File: projects/packages/wordads/tools/webpack.dashboard.config.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			jetpackWebpackConfig.TranspileRule( {
+				includeNodeModules: [ '@automattic/jetpack-' ],
+			} ),
+
+			// Handle CSS.
+			jetpackWebpackConfig.CssRule( {
+				extensions: [ 'css', 'sass', 'scss' ],
+				extraLoaders: [
+					{
+						loader: 'postcss-loader',
```
Comment (kangzj) [inline projects/packages/wordads/tools/webpack.dashboard.config.js:63]: We discovered an issue here, the `postcss-custom-properties.preserve` is set to false, which leads to css variables are not compiled into the target file. Currently we removed `postcss-loader`.

File: projects/plugins/jetpack/class.jetpack.php [hunk 2]
```diff
@@ -588,69 +593,7 @@ class_exists( 'Jetpack_Sitemap_Manager' )
...
-			 *
-			 * @since 1.9.0
-			 *
-			 * @param string $module Module slug.
-			 */
-			do_action( "jetpack_deactivate_module_$module", $module );
-		}
-
-		return $success;
+		return ( new Modules() )->update_active( $modules );
```
Comment (kangzj) [inline projects/plugins/jetpack/class.jetpack.php:596]: 👍

File: tools/cli/skeletons/common/package.json [hunk 3]
```diff
@@ -14,10 +14,10 @@
 	"license": "GPL-2.0-or-later",
 	"author": "Automattic",
 	"scripts": {
-		"build": "echo 'Not implemented.",
-		"build-js": "echo 'Not implemented.",
-		"build-production": "echo 'Not implemented.",
-		"build-production-js": "echo 'Not implemented.",
+		"build": "echo 'Not implemented.'",
```
Comment (kraftbj) [inline tools/cli/skeletons/common/package.json:17]: I'll fix all of these in a separate PR to help clean it out of this one.

File: projects/packages/wordads/tools/webpack.dashboard.config.js (hunk 1 above)
Comment (gravityrail) [reply projects/packages/wordads/tools/webpack.dashboard.config.js]: Fixed, thanks

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
r243716-wpcom

**Pull request merged**